### PR TITLE
Fix knative installation failure in e2e test

### LINKED ISSUE #2730: Knative installation keeps failing in e2e tests
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
The e2e tests are failing every now and then while running the knative installation step, more specifically while patching the configmap. A solution has to be provided so that the installation completes successfully using some kind of retry mechanism.

**What did you expect to happen:**
All e2e tests to run without any issues.

**Environment:**

e2e environment

### PR DESCRIPTION
**Pull request by andyi2it**
**What this PR does / why we need it**:
Add a sleep.
Add a simple retry mechanism in case of failure.

* Fixes issue with knative installation/patch during e2e*

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2730 #2731

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Feature/Issue validation/testing**:

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Has code been commented, particularly in hard-to-understand areas?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -324,6 +324,9 @@ jobs:
       - uses: actions/setup-go@v2
         with:
           go-version: '1.17.3'
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
```
Comment (rachitchauhan43) [inline .github/workflows/e2e-test.yml:329]: @andyi2it : Curious to know how did you figure out from the [logs](https://github.com/kserve/kserve/actions/runs/4315581912/jobs/7531712384) that it was python version issue.

Would you mind sharing how you figured this out?

Comment (andyi2it) [reply .github/workflows/e2e-test.yml:329]: @rachitchauhan43 
Its not from the e2e logs alone.
I ran it locally, and compared the differences. My local had an earlier version of python and it was working.
So, I tried pinning the version. 
However, the most accurate detail is that, the portforward library was the one that was breaking the e2e.
The newer versions of python uses the newer portforward library by default and hence the error.

Given that the root cause is identified, we can remove pinning python to 3.9. 
However, I think we should pin python to the max version we support. Will create a separate issue for that.

Comment (rachitchauhan43) [reply .github/workflows/e2e-test.yml:329]: @andyi2it : This is really awesome work. Thanks for this fix 👍

File: test/scripts/gh-actions/setup-deps.sh [hunk 2]
```diff
@@ -71,11 +71,16 @@ for i in 1 2 3 ; do kubectl apply -k test/overlays/knative && break || sleep 15;
...
-echo "Add knative hpa..."
+# echo "Add knative hpa..."
 # kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.0.0/serving-hpa.yaml
 
 # Skip tag resolution for certain domains
-kubectl patch cm config-deployment --patch '{"data":{"registries-skipping-tag-resolving":"nvcr.io,index.docker.io"}}' -n knative-serving
+# sleep to avoid knative webhook timeout error
+sleep 5
+# Retry if configmap patch fails
+for i in 1 2 3; do
```
Comment (rachitchauhan43) [inline test/scripts/gh-actions/setup-deps.sh:81]: @andyi2it : This is to address which test failure ?

**Review by rachitchauhan43 — APPROVED** [review]

File: test/scripts/gh-actions/setup-deps.sh (hunk 2 above)
Comment (andyi2it) [reply test/scripts/gh-actions/setup-deps.sh:81]: This one is for #2730.
This PR fixes both #2730 and #2731

**Pull request merged**